controller/api/destination/server_test.go

```diff
@@ -158,6 +159,110 @@ func TestGet(t *testing.T) {
 		}
 	})
 
+	t.Run("Return endpoint opaque protocol controlled by a server", func(t *testing.T) {
```

Comment:
can we add the same test for external workloads ?

Reply:
great question.  I tried this and then spent forever racking my brain trying to figure out why it wasn't working.  it turns out the server_tests all run with the endpoint_watcher in Endpoints mode, not EndpointSlices mode, and external workloads are not compatible with Endpoints.  

Given that EndpointSlices mode is the default, we'll want to rewrite these tests to use it, but not in this PR.